models/block.go

```diff
@@ -0,0 +1,90 @@
+package models
+
+import (
+	"encoding/json"
+
+	"github.com/libsv/go-bt/v2"
+)
+
+// BlockDecodeHeader model.
+type BlockDecodeHeader struct {
+	Txs []string `json:"tx"`
+	BlockHeader
+}
+
+// Block model.
+type Block struct {
+	Txs bt.Txs `json:"tx"`
+	BlockHeader
+}
+
+// UnmarshalJSON unmarshal response.
+func (b *Block) UnmarshalJSON(bb []byte) error {
+	bj := struct {
+		Txs json.RawMessage `json:"tx"`
+		BlockHeader
+	}{}
+	if err := json.Unmarshal(bb, &bj); err != nil {
+		return err
+	}
+
+	var txs bt.Txs
+	if err := json.Unmarshal(bj.Txs, txs.NodeJSON()); err != nil {
+		return err
+	}
+
+	b.Txs = txs
+	b.BlockHeader = bj.BlockHeader
+	return nil
+}
+
+// BlockHeader model.
+type BlockHeader struct {
```

Comment:
do you wanna use the go-bc block header here for some of the fields like you did with the tx output?

Reply:
This struct here has extra fields, if you're happy to lose them then yeah I can move this over

Reply:
i meant have the bc header plus these extra fields (like you did here: https://github.com/libsv/go-bn/pull/1/files#diff-6d5301c97541b896843d5363919d90686ed3489ab0ced2d2977848ff6daafb74R17)

Reply:
Lol I didn't even remember doing that. I'll move that there, good find